src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -139,8 +140,13 @@ internal partial class Binder
                     else
                     {
                         type = BindType(typeSyntax, diagnostics);
-                        ParameterHelpers.CheckParameterModifiers(p, diagnostics, parsingFunctionPointerParams: false, parsingLambdaParams: true);
-                        refKind = ParameterHelpers.GetModifiers(p.Modifiers, out _, out _, out _, out scope);
+                        ParameterHelpers.CheckParameterModifiers(p, diagnostics, parsingFunctionPointerParams: false, parsingLambdaParams: true,
+                            parsingAnonymousMethod: syntax.Kind() == SyntaxKind.AnonymousMethodExpression);
+                        refKind = ParameterHelpers.GetModifiers(p.Modifiers, out _, out var paramsKeyword, out _, out scope);
+                        if (paramsKeyword.Kind() != SyntaxKind.None)
+                        {
+                            hasParamsArray = true;
```

Comment:
Should `hasParamsArray` be set only if this is the last parameter?

Reply:
I guess it's better, so I added the check: 9259b6e6e45105b093b8f5a9bd5f3d1b61e36f03. Although I don't think it's strictly necessary as the binding will fail anyway if `params` is not at the last parameter (at least when that's implemented in the next PR).

Also, it occurred to me that a parameter symbol should probably have `IsParams` set correctly even if it's not the last one, so I changed that: 4a707295aebcfa0e904f604ffd2f998710d4c167